File: algorithms/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nums_len = len(nums)
        if nums_len < 2:
            return nums_len
        p, q = 0, 1
        while q < nums_len:
            if nums[p] == nums[q]:
                q += 1
            else:
                p += 1
                nums[p] = nums[q]
                q += 1
        return p + 1

File: algorithms/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
	length := len(nums)
	if length < 2 {
		return length
	}
	p, q := 0, 1
	for q < length {
		if nums[p] != nums[q] {
			p++
			nums[p] = nums[q]
		}
		q++
	}
	return p + 1
}
